packages/react-dom/src/server/ReactPartialRenderer.js

```diff
@@ -838,6 +858,10 @@ class ReactDOMServerRenderer {
 
     const prevThreadID = currentThreadID;
     setCurrentThreadID(this.threadID);
+    const prevUniqueID = currentUniqueID;
+    setCurrentUniqueID(this.uniqueID);
```

Comment:
You pass this value and increment it but you don't actually ever update `this.uniqueID`. For the synchronous renderToString this won't show up (except maybe for nested calls which we have a open issue about). However for two streaming renders that read multiple times, it'll reset to zero between each read.

You might want to write a unit test that catches this as since it could be pretty bad if we regress on it.

As a solution I'd just pass `this` instead. setCurrentRenderer(this) and read both the uniqueID, threadID and prefix from the renderer directly.